src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -9,15 +9,52 @@ import * as vscode from 'vscode'
 import { fileExists } from '../../filesystemUtilities'
 import { DefaultSamCliTaskInvoker, SamCliTaskInvoker } from './samCliInvoker'
 
+export interface SamCliLocalInvokeInvocationArguments {
+    // Todo : comments incoming...
+    templateResourceName: string,
+    templatePath: string,
+    eventPath: string,
+    environmentVariablePath: string,
+    debugPort?: string,
+    invoker: SamCliTaskInvoker,
+    useContainer?: boolean,
+    dockerNetwork?: string,
+    skipPullImage?: boolean,
+}
+
 export class SamCliLocalInvokeInvocation {
-    public constructor(
-        private readonly templateResourceName: string,
-        private readonly templatePath: string,
-        private readonly eventPath: string,
-        private readonly environmentVariablePath: string,
-        private readonly debugPort?: string,
-        private readonly invoker: SamCliTaskInvoker = new DefaultSamCliTaskInvoker()
+    private readonly templateResourceName: string
```

Comment:
What do you think about making `args` a field, rather than repeating the fields from `SamCliLocalInvokeInvocationArguments`?

Reply:
I do like having the fields here, because we "transform" out how they are intended to be used. In this case, we want useContainer and skipPullImage to be non-undefined